samplePDF/samplePDFFDBase.cpp

```diff
@@ -806,171 +800,110 @@ void samplePDFFDBase::CalcXsecNormsBins(int iSample){
-			  //}
-			  IsSelected = false;
-			  continue;
-			}
-			else if (ReturnKinematicParameter((*it).KinematicVarStr[iKinematicParameter], iSample, iEvent) > (*it).Selection[iKinematicParameter][1]) {
-			  IsSelected = false;
-			  continue;
-			}
-		  } 
-		}
-		//Need to then break the event loop 
-		if(!IsSelected){
-		  continue;
-		}
+    if (XsecCov) {
+      for (std::vector<XsecNorms4>::iterator it = xsec_norms.begin(); it != xsec_norms.end(); ++it) {
+	// Skip oscillated NC events
+	// Not strictly needed, but these events don't get included in oscillated predictions, so
+	// no need to waste our time calculating and storing information about xsec parameters
+	// that will never be used.
+	if (fdobj->isNC[iEvent] && fdobj->signal) {continue;} //DB Abstract check on MaCh3Modes to determine which apply to neutral current
+	
+	//Now check that the target of an interaction matches with the normalisation parameters
+	bool TargetMatch=false;
+	//If no target specified then apply to all modes
+	if ((*it).targets.size()==0) {
+	  TargetMatch=true;
+	} else {
+	  for (unsigned iTarget=0;iTarget<(*it).targets.size();iTarget++) {
+	    if ((*it).targets.at(iTarget)== *(fdobj->Target[iEvent])) {
+	      TargetMatch=true;
+	    }
+	  }
+	}
+	if (!TargetMatch) {continue;}
+	
+	//Now check that the neutrino flavour in an interaction matches with the normalisation parameters
+	bool FlavourMatch=false;
+	//If no mode specified then apply to all modes
+	if ((*it).pdgs.size()==0) {
+	  FlavourMatch=true;
+	} else {
+	  for (unsigned iPDG=0;iPDG<(*it).pdgs.size();iPDG++) {
+	    if ((*it).pdgs.at(iPDG)== fdobj->nupdg) {
+	      FlavourMatch=true;
+	    }
+	  }
+	}
+	if (!FlavourMatch){continue;}
+	
+	//Now check that the unoscillated neutrino flavour in an interaction matches with the normalisation parameters
+	bool FlavourUnoscMatch=false;
+	//If no mode specified then apply to all modes
+	if ((*it).preoscpdgs.size()==0) {
+	  FlavourUnoscMatch=true;
+	} else {
+	  for (unsigned iPDG=0;iPDG<(*it).preoscpdgs.size();iPDG++) {
+	    if ((*it).preoscpdgs.at(iPDG) == fdobj->nupdgUnosc) {
+	      FlavourUnoscMatch=true;
+	    }
+	  }
+	}
+	if (!FlavourUnoscMatch){continue;}
+	
+	//Now check that the mode of an interaction matches with the normalisation parameters
+	bool ModeMatch=false;
+	//If no mode specified then apply to all modes
+	if ((*it).modes.size()==0) {
+	  ModeMatch=true;
+	} else {
+	  for (unsigned imode=0;imode<(*it).modes.size();imode++) {
+	    if ((*it).modes.at(imode)== *(fdobj->mode[iEvent])) {
+	      ModeMatch=true;
+	    }
+	  }
+	}
+	if (!ModeMatch) {continue;}
+	
+	//Now check whether the norm has kinematic bounds
+	//i.e. does it only apply to events in a particular kinematic region?
+	bool IsSelected = true;
+	if ((*it).hasKinBounds) {
```

Comment:
its so minor, but I really don't understand the `(*it).`, why not just use the syntax that everyone understands/learns in C++ 101, `it->hasKinBounds`?